app/src/lib/api.ts

```diff
@@ -280,6 +307,91 @@ function getNextPagePath(response: Response): string | null {
   return null
 }
 
+/**
+ * Parses the 'next' Link header from GitHub using
+ * `getNextPagePathFromLink`. Unlike `getNextPagePathFromLink`
+ * this method will attempt to double the page size when
+ * the current page index and the page size allows for it
+ * leading to a ramp up in page size.
+ *
+ * This might sound confusing, and it is, but the primary use
+ * case for this is when retrieving updated PRs. By specifying
+ * an initial page size of, for example, 10 this method will
+ * increase the page size to 20 once the second page has been
+ * loaded. See the table below for an example. The ramp-up
+ * will stop at a page size of 100 since that's the maximum
+ * that the GitHub API supports.
+ *
+ * ```
+ * |-----------|------|-----------|-----------------|
+ * | Request # | Page | Page size | Retrieved items |
+ * |-----------|------|-----------|-----------------|
+ * | 1         | 1    | 10        | 10              |
+ * | 2         | 2    | 10        | 20              |
+ * | 3         | 2    | 20        | 40              |
+ * | 4         | 2    | 40        | 80              |
+ * | 5         | 2    | 80        | 160             |
+ * | 6         | 3    | 80        | 240             |
+ * | 7         | 4    | 80        | 320             |
+ * | 8         | 5    | 80        | 400             |
+ * | 9         | 2    | 100       | 500             |
```

Comment:
You're absolutely right :sparkles:. This could have been bad but luckily the current maximum retrieved items prevent us from ever getting into this place (which is likely contributing to why I didn't see this). I've fixed it and also codified the algorithm in tests to verify the correct behavior.